test/test_utils.jl

```diff
@@ -251,6 +251,28 @@ coords_on_faces(x, ::Serendipity{RefHexahedron, 2}) =
 check_equal_or_nan(a::Any, b::Any) = a==b || (isnan(a) && isnan(b))
 check_equal_or_nan(a::Union{Tensor, Array}, b::Union{Tensor, Array}) = all(check_equal_or_nan.(a, b))
 
+# Hypercube is simply ⨂ᵈⁱᵐ Line :)
+sample_random_point(::Type{Ferrite.RefHypercube{ref_dim}}) where {ref_dim} = Vec{ref_dim}(2.0 .* rand(Vec{ref_dim}) .- 1.0)
+# Dirichlet type sampling
+function sample_random_point(::Type{Ferrite.RefSimplex{ref_dim}}) where {ref_dim} 
```

Comment:
Looks beautiful :) Posting here as future reference
```julia
julia> using UnicodePlots

julia> function sample_random_point()
           ξ = rand(3)
           ξₜ = -log.(ξ)
           return ξₜ[1:2] / sum(ξₜ)
       end
sample_random_point (generic function with 1 method)

julia> x = zeros(1000); y = zeros(1000);

julia> for i in 1:length(x)
           x[i], y[i] = sample_random_point()
       end

julia> scatterplot(x, y)
```

![image](https://github.com/Ferrite-FEM/Ferrite.jl/assets/19473844/08ab1bd2-db26-442c-9f1c-3d7327d6e7de)

Reply:
Any feelings regarding putting this into Ferrite instead of docs only? I think I have a legit use case for this already in the point finder (within its current design at least).

Reply:
As procrastination I looked at the initial guessing a bit, and perhaps a reproducible way is better? For testing I think this is nice, but for actual computations I'm not currently in favor of introducing randomness in the code. 

Here is a code-snippet with some trial changes, where the guess is calculated based on the closest point on a line between the two closest cell coordinates. 
The main internally breaking change would be that `reference_coordinates` should return a `Tuple` instead of `Vector` to avoid allocations...
```julia
# Find the two closest cell coordinates to the target coordinate
function find_two_closest(x_cell, x_target)
    i1 = 0
    i2 = 0
    d1 = typemax(eltype(x_target))
    d2 = d1
    for (i, x) in pairs(x_cell)
        d = norm(x-x_target)
        if d < d1
            i2 = i1
            d2 = d1
            i1 = i
            d1 = d
        end
    end
    return i1, i2, d2/(d1+d2)
end

# Define this function as a simple test for testing `QuadraticQuadrilateral`
function reference_coords()
    return (Vec{2, Float64}((-1.0, -1.0)),
            Vec{2, Float64}(( 1.0, -1.0)),
            Vec{2, Float64}(( 1.0,  1.0)),
            Vec{2, Float64}((-1.0,  1.0)),
            Vec{2, Float64}(( 0.0, -1.0)),
            Vec{2, Float64}(( 1.0,  0.0)),
            Vec{2, Float64}(( 0.0,  1.0)),
            Vec{2, Float64}((-1.0,  0.0)),
            Vec{2, Float64}(( 0.0,  0.0)))
end

# See https://discourse.julialang.org/t/finding-the-value-of-a-field-at-a-spatial-location-in-juafem/38975/2
# TODO: should we make iteration params optional keyword arguments?
function find_local_coordinate(interpolation, cell_coordinates::Vector{V}, global_coordinate::V) where {dim, T, V <: Vec{dim, T}}
    n_basefuncs = getnbasefunctions(interpolation)
    @assert length(cell_coordinates) == n_basefuncs
    i1, i2, w = find_two_closest(cell_coordinates, global_coordinate)
    ξ1 = reference_coords()[i1]
    ξ2 = reference_coords()[i2]
    local_guess = ξ1*w + ξ2*(1-w)
    #local_guess = zero(V)
    max_iters = 10
    tol_norm = 1e-10
    converged = false
    local niter
    for iter in 1:max_iters
        global_guess = zero(V)
        J = zero(Tensor{2, dim, T})
        for j in 1:n_basefuncs
            dNdξ, N = shape_gradient_and_value(interpolation, local_guess, j)
            global_guess += N * cell_coordinates[j]
            J += cell_coordinates[j] ⊗ dNdξ
        end
        residual = global_guess - global_coordinate
        if norm(residual) <= tol_norm
            converged = true
            niter = iter
            break
        end
        local_guess -= inv(J) ⋅ residual
    end
    return converged, local_guess, niter
end
```

Reply:
For this case we can introduce a function `reference_coordinates!` which just writes into a buffer. :)